manifests/resource/upstream/member.pp

```diff
@@ -40,11 +42,17 @@
   Enum['present', 'absent'] $ensure = 'present',
   Integer $port                     = 80,
   $upstream_fail_timeout  = '10s',
+  Enum['http', 'stream'] $upstream_context  = 'http',
 ) {
 
+  $conf_dir_real = $upstream_context ? {
+    'stream' => 'conf.stream.d',
+    default  => 'conf.d',
+  }
+
   # Uses: $server, $port, $upstream_fail_timeout
   concat::fragment { "${upstream}_upstream_member_${name}":
-    target  => "${::nginx::conf_dir}/conf.d/${upstream}-upstream.conf",
+    target  => "${::nginx::conf_dir}/${conf_dir_real}/${upstream}-upstream.conf",
```

Comment:
i think it's worth adding an rspec test for this

Reply:
@cova-fe is this something you're still able to work on?

Reply:
Yes, I should be able to do it :)